scrapper.py

```diff
@@ -21,43 +40,160 @@ class IncorrectNumberOfArticlesError(Exception):
     """
 
 
+def _clean_text(text):
+    return re.sub(r"[\n\t ]+", " ", text).strip()
+
+
+def _get_page(link):
+    print("attempting to connect to %s", link)
+    #time.sleep(random.uniform(2.0, 4.0))
+    headers = Headers().generate()
+    print(headers)
+    response = requests.get(link,
+                            headers=headers,
+                            timeout=5)
+    if not response.ok:
+        return None
+    return BeautifulSoup(response.text, "html.parser")
+    # except requests.exceptions.ConnectionError:
+    #    print("failed to connect to %s. trying again", link)
+        # return _get_page(link)
+    #    raise ServerThrottledError
+    #except requests.exceptions.ReadTimeout:
+        #print("timed out connecting to %s. trying again", link)
+        #return _get_page(link)
+
+
 class Crawler:
```

Comment:
as soon as you are doing it for 10 there should be also implementation for a recursive crawler which is a bit more complicated than a "traditional" one. without it you will not be able to pretend for 10

Reply:
It has now been implemented